optuna/pruners/successive_halving.py

```diff
@@ -128,6 +146,25 @@ def prune(self, study, trial):
 
             rung += 1
 
+    def _estimate_min_resource(self, trials):
+        # type: (Optional[List[FrozenTrial]]) -> None
+
+        if trials is None:
+            return
+
+        complete_trials = [t for t in trials if t.state == TrialState.COMPLETE]
+
+        if not complete_trials:
+            return
+
+        # get the number of steps and divide it by 100.
+        complete_trial = complete_trials[0]  # type: FrozenTrial
+        last_step = complete_trial.last_step
+        if last_step is not None:
+            hundredth = last_step // 100
+            self._min_resource = hundredth if hundredth > 0 else 1
+            assert self._min_resource > 0
```

Comment:
It's obvious from the above line (L165) that this condition holds, so I think we can remove this assertion.